vnext/Microsoft.ReactNative/Modules/AlertModule.cpp

```diff
@@ -16,6 +16,16 @@
 namespace Microsoft::ReactNative {
 
 void Alert::showAlert(ShowAlertArgs const &args, std::function<void(std::string)> result) noexcept {
+  pendingAlerts.push({args, result});
+  if (pendingAlerts.size() == 1) {
```

Comment:
I think we have a potential race condition here. If this is a default thread affinity module, we're pushing and checking the alert count on the JS thread, but decrementing and popping on the UI thread. 

It should be easier to reason about if you keep queue state on a single thread, so always dispatching to the UI thread before doing stuff with the state here.